lisa/analysis/rta.py

```diff
@@ -15,187 +15,493 @@
+            if not evt.startswith('rtapp_'):
+                continue
+            fnc_name = "df_{}".format(evt)
+            df_getter = getattr(self.trace.analysis.rta, fnc_name)
+            df = df_getter()
+            for pid,name in df[['__pid', '__comm']].drop_duplicates().values:
+                task_ids.add(TaskID(pid, name))
+        return list(task_ids)
+
+    @property
+    def rtapp_tasks(self):
+        return self._get_rtapp_tasks()
+
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+
+    ###########################################################################
+    # rtapp_main events related methods
+    ###########################################################################
+
+    @requires_events('rtapp_main')
+    def df_rtapp_main(self):
+        """
+        Dataframe of events generated by the rt-app main task.
+
+        :returns: a :class:`pandas.DataFrame` with:
+
+          * A ``__comm`` column: the actual rt-app trace task name
+          * A ``__cpu``  column: the CPU on which the task was running at event
+                                 generation time
+          * A ``__line`` column: the ftrace line numer
+          * A ``__pid``  column: the PID of the task
+          * A ``data``   column: the data corresponding to the reported event
+          * An ``event`` column: the event generated
+
+        The ``event`` column can report these events:
+
+          * ``start``: the start of the rt-app main thread execution
+          * ``end``: the end of the rt-app main thread execution
+          * ``clock_ref``: the time rt-app gets the clock to be used for logfile entries
+
+        The ``data`` column reports:
+
+          * the base timestamp used for logfile generated event for the ``clock_ref`` event
+          * ``NaN`` for all the other events
+
+        """
+        return self.trace.df_events('rtapp_main')
+
+    @property
+    @df_rtapp_main.used_events
+    def rtapp_window(self):
+        """
+        Return the time range the rt-app main thread executed.
+
+        :returns: a tuple(start_time, end_time)
+        """
+        df = self.df_rtapp_main()
+        return (
+            df[df.event == 'start'].index.values[0],
+            df[df.event == 'end'].index.values[0])
+
+    @property
+    @df_rtapp_main.used_events
+    def rtapp_reftime(self):
+        """
+        Return the tuple representing the ``kernel`` and ``user`` timestamp.
+
+        RTApp log events are generated based on a reference timestamp.
+        This method allows to know which trace timestamp corresponds to the
+        logfile generated timestamps.
+
+        :returns: a namedtuple `RefTime` reporting ``kernel`` and ``user``
+                  timestamps.
+        """
+        RefTime = namedtuple("RefTime", ['kernel', 'user'])
```

Comment:
RefTime definition should be moved outside the function body (either in the class or in the global scope), so that the name it will claim to have can actually be "resolved" to get the type. (otherwise this might bite someone trying to pickle instances for example)